includes/class-omise-money.php

```diff
@@ -0,0 +1,87 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var float
+	 */
+	protected $amount;
+
+	/**
+	 * @var string
+	 */
+	protected $currency;
+
+	/**
+	 * @var array
+	 */
+	private $subunit_multiplier = array(
```

Comment:
Would it not be easier just to store ones that aren't 100? and if not in the list, just default to 100?

Reply:
Currency support is handled elsewhere. I don't think this code should be dealing with that

Reply:
> Would it not be easier just to store ones that aren't 100? and if not in the list, just default to 100?
This method could be simplify to 
```php
$subunit_multiplier = [ 'jpy' => 1 ];
$multiplier         = isset( $subunit_multiplier[ $order->currency ] ) ? $subunit_multiplier[ $order->currency ] : 100;
```

I chose another way around as showing all currencies is just more clearer to read (might not the shortest and smartest code, but I account for the readability on this one).

However, there might be another way I didn't think of. Wanna share some snippet of code of your thought?

Reply:
Pretty much what you showed above. It is much simpler, and reduces future maintenance burden (it will rarely need to change when new currencies are added)

Reply:
I agreed on the simplicity. But still, wanna stand for the readability in this case.

The above one could be easy to maintain, but raising more questions at the same time.
It is not just benefit us but for anyone that read the code without context of what currency Omise supported.

How about just leave it as it is for now, and I will change to the short one later if it's really a burden for maintenance?

Reply:
The currencies that Omise support aren't really relevant to what should be very simple conversion functions - which is all that we need here

Reply:
I understand your point of view and idea for the simplicity.
To me, it's just choosing between explicit way or smart way and I normally tend to go with an explicit way if possible.

Is there any concrete benefit to switch to a smarter way? (besides maintenance – automatically support for a new currency that multiply 100 for its subunit by default – which I will be willingly to change if it comes to really a burden for a maintenance later on),

Reply:
(many parts of the project could be re-written into just 1 or 2 lines and it would function just the same, yet, we break the line, we add a meaningless variable and assign a value to it just we can pass it to another function at the very next line, etc. I account that for readability sake – same as this code)